torch/csrc/autograd/source_debug_info.cpp

```diff
@@ -0,0 +1,45 @@
+#include  <torch/csrc/autograd/graph_debug_info.h>
+#include  <torch/csrc/autograd/source_debug_info.h>
+
+namespace torch { namespace autograd { namespace profiler {
+std::mutex module_source_debug_info_lock;
```

Comment:
global locks are evil, especially in multithreaded environment (e.g. in predictor). Can we instead attach debug info at Function level or rely on shared_ptrs?

Reply:
@dzhulgakov, this lock is used only at the time of code creation. Is it still a concern?

Reply:
Code creation is called under a compilation mutex in GraphExecutor: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/graph_executor.cpp#L532

Can we rely on that instead?